ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsService.scala

```diff
@@ -304,6 +305,19 @@ class ContractsService(
   private def lfValueToJsValue(a: LfValue): Error \/ JsValue =
     \/.fromTryCatchNonFatal(LfValueCodec.apiValueToJsValue(a)).leftMap(e =>
       Error('lfValueToJsValue, e.description))
+
+  private def resolveTemplateIds(xs: Set[domain.TemplateId.OptionalPkg])
+    : (Set[domain.TemplateId.RequiredPkg], Set[domain.TemplateId.OptionalPkg]) = {
+
+    val z = (Set.empty[domain.TemplateId.RequiredPkg], Set.empty[domain.TemplateId.OptionalPkg])
+
+    xs.toList.foldLeft(z) { (b, a) =>
```

Comment:
```suggestion
    xs partitionMap {
      resolveTemplateId(_) toLeftDisjunction a
    }
```

putting this in an appropriate place in utils

https://github.com/digital-asset/daml/blob/a32e1056397c8ce157c81d9a61746261f3c22d75/ledger-service/http-json/src/main/scala/com/digitalasset/http/WebSocketService.scala#L44-L57

Reply:
I don't think `WebSocketService` is an appropriate place for this util. My change has nothing to do with the WebSocket.

Reply:
It's not an appropriate place; put it wherever you think is good.

Reply:
too much magic....
```
ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsService.scala:315: error: value partitionMap is not a member of Set[com.digitalasset.http.domain.TemplateId.OptionalPkg]
    xs.partitionMap {
       ^
ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsService.scala:313: warning: Unused import
    import ContractsService._
```

Reply:
`toSeq` helped